### PR TITLE
(DOCSP-11659): MongoDB Data Access Web Long Doc

### PR DESCRIPTION
**Pull request by mongomoe**
jira: https://jira.mongodb.org/browse/DOCSP-11659
stage: https://docs-mongodbcom-staging.corp.mongodb.com/realm/mohammad.hunan/DOCSP-11659-MongoDB-Data-Access-Web/web/mongodb.html

### CODE REVIEW

File: source/web/mongodb.txt [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+
+Insert a Single Document
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can insert a single document using :js-sdk:`collection.insertOne() <Realm.RemoteMongoDBCollection.html#insertOne>`.
+
+The following snippet inserts a single document describing a "lily of the
+valley" plant into a
+:ref:`collection of documents that describe plants for sale in a group of stores
```
Comment (ccho-mongodb) [inline source/web/mongodb.txt:47]: I don't think these links are working. Do you need to specify the ref label by the section? That should fix all the instances I think.

File: source/web/mongodb.txt [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+
+Read Documents
+--------------
+
+These code snippets demonstrate how to read data stored in a MongoDB
+collection from a mobile application. Read operations use :manual:`query
+filters </core/document/#document-query-filter>` to specify which
+documents to return from the database. Read operations return a
+:mdn:`Promise <Web/JavaScript/Reference/Global_Objects/Promise>` that
+resolves to either a single matched document (in the case of
```
Comment (ccho-mongodb) [inline source/web/mongodb.txt:128]: Nitpick: "either" is usually used for exactly two items. Here, you list three, so I would use "one of the following: " or something similar.

File: source/web/mongodb.txt [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+
+.. code-block:: shell
+
+   [ { _id: 5f1f4266e0ef86239b6c416c, name: 'venus flytrap', sunlight: 'full', color: 'white', type: 'perennial', _partition: 'Store 42' }, { _id: 5f1f4266e0ef86239b6c416d, name: 'sweet basil', sunlight: 'partial', color: 'green', type: 'annual', _partition: 'Store 42' }, { _id: 5f1f4266e0ef86239b6c416e, name: 'thai basil', sunlight: 'partial', color: 'green', type: 'perennial', _partition: 'Store 42' }, { _id: 5f1f4266e0ef86239b6c416f, name: 'helianthus', sunlight: 'full', color: 'yellow', type: 'annual', _partition: 'Store 42' }, { _id: 5f1f4500e0ef86239b6d7ae7, name: 'wisteria lilac', sunlight: 'partial', color: 'purple', type: 'perennial', _partition: 'Store 42' }, { _id: 5f1f4500e0ef86239b6d7ae8, name: 'daffodil', sunlight: 'full', color: 'yellow', type: 'perennial', _partition: 'Store 42' } ]
+
+Count Documents in the Collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can count documents in a collection using
+:js-sdk:`collection.count() <Realm.RemoteMongoDBCollection.html#findOne>`.
```
Comment (ccho-mongodb) [inline source/web/mongodb.txt:189]: Is this the correct link? I see findOne, not count(). I think you meant:
Realm.RemoteMongoDBCollection.html#count

File: source/web/mongodb.txt [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+
+.. _web-mongodb-deleteOne:
+
+Delete a Single Document
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can delete a single document from a collection using
+:js-sdk:`collection.deleteOne() <Realm.RemoteMongoDBCollection.html#deleteOne>`.
+
+The following snippet deletes one document in a
```
Comment (ccho-mongodb) [inline source/web/mongodb.txt:339]: Nitpick: "deletes... from a"

File: source/includes/steps-crud-snippets-web.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+---
+title: Instantiate a MongoDB Collection Handle
+ref: instantiate-a-mongodb-collection-handle
+level: 4
+content: |
+
+  .. code-block:: javascript
+     :emphasize-lines: 15,16
+
+     const appId = "<your App ID>"; // Set Realm app ID here.
```
Comment (mongomoe) [inline source/includes/steps-crud-snippets-web.yaml:36]:
```suggestion
     const appId = "<your App ID>"; // Set Realm App ID here.
```

[the remark this answers is not in the document]
Comment (mongomoe) [reply source/includes/steps-crud-snippets-web.yaml:36]: 👍

File: source/includes/steps-crud-snippets-web.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+level: 4
+content: |
+
+  .. code-block:: javascript
+     :emphasize-lines: 15,16
+
+     const appId = "<your App ID>"; // Set Realm App ID here.
+     const appConfig = {
+       id: appId,
+       timeout: 10000, // timeout in milliseconds
```
Comment (mongomoe) [inline source/includes/steps-crud-snippets-web.yaml:39]:
```suggestion
       timeout: 10000, // timeout in number of milliseconds
```

File: source/web.txt [hunk 7]
```diff
@@ -33,8 +33,10 @@ TypeScript applications.
    Manage Email/Password Users </web/manage-email-password-users>
    Create & Manage User API Keys </node/create-manage-api-keys>
    Work with Multiple Users </web/work-with-multiple-users>
+   MongoDB Data Access <web/mongodb>
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/web.txt:36]: For this PR I want to keep the doc title consistent with the titles for the other SDKs, but we can reevaluate later

File: source/web/mongodb.txt (hunk 1 above)
Comment (mongomoe) [reply source/web/mongodb.txt:48]: Good catch - I added the ref: ".. _web-mongodb-example-dataset:" now

File: source/web/mongodb.txt (hunk 2 above)
Comment (mongomoe) [inline source/web/mongodb.txt:129]:
```suggestion
resolves to one of the following: a single matched document (in the case of
```

Comment (mongomoe) [reply source/web/mongodb.txt:129]: 👍

File: source/web/mongodb.txt (hunk 3 above)
Comment (mongomoe) [reply source/web/mongodb.txt]: 👍

File: source/web/mongodb.txt [hunk 8]
```diff
@@ -0,0 +1,431 @@
...
```
Comment (mongomoe) [inline source/web/mongodb.txt:340]:
```suggestion
The following snippet deletes one document from a
```

**Pull request merged**